net/utils.go

```diff
@@ -70,7 +70,7 @@ func GenericTLSWrap(conn *net.TCPConn, cfg *tls.Config, tFunc TLSWrapperFunc) (*
 					continue
 				}
 			}
-			return nil, err
+			continue
```

Comment:
@doodles526 can you walk me thru this change of behavior for doing the TLS handshake? The way I understand it, we'll keep attempting a handshake until we get one or we weren't able to set the deadline again (because the conn is dead or similar). That suggest to me that the loop is going to be pretty tight. At the very top of the wormhole client we have a loop that that tries to connect, backs off and tries again. Would it make more sense to just error out here or at the very least limit the handshake to _n_ attempts?

Reply:
Yeah, it would probably make sense to either err-out or have an exponential back-off with n attempts. In a near-future PR I think we can fix some of these negotiation issues with better messaging structures. But for now, just err out and let the client retry a new connection like you said